bees/horizonboxbee/horizonboxbee.go

```diff
@@ -0,0 +1,162 @@
+/*
+ *    Copyright (C) 2017 Dominik Schmidt
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU Affero General Public License as published
+ *    by the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Authors:
+ *      Dominik Schmidt <dev@dominik-schmidt.de>
+ */
+
+package horizonboxbee
+
+import (
+	"fmt"
+	"github.com/PuerkitoBio/goquery"
+	"github.com/muesli/beehive/bees"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type HorizonBoxBeeState struct {
+	online bool
+	ip     string
+}
+
+type HorizonBoxBee struct {
+	bees.Bee
+
+	address  string
+	user     string
+	password string
+
+	State     HorizonBoxBeeState
+	eventChan chan bees.Event
+}
+
+func (mod *HorizonBoxBee) poll() {
+	// Form data
+	v := url.Values{}
+	v.Set("username", mod.user)
+	v.Set("password", mod.password)
+	v.Set("page", "login")
+
+	req, err := http.NewRequest("POST", "http://"+mod.address+"/cgi-bin/sendResult.cgi?section=login", strings.NewReader(v.Encode()))
+	if err != nil {
+		mod.LogErrorf("http.NewRequest() error: %v\n", err)
+		return
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded; charset=utf-8")
+
+	cookieJar, _ := cookiejar.New(nil)
+	c := &http.Client{
+		Jar: cookieJar,
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		mod.LogErrorf("Could not connect to: %s", mod.address, err)
```

Comment:
This should probably read `mod.LogErrorf("Could not connect to %s: %v", mod.address, err)`